lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -822,7 +822,7 @@ public function count($criteria = array())
             ? $this->expandCriteriaParameters($criteria)
             : $this->expandParameters($criteria);
 
-        return $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
+        return (int) $this->conn->executeQuery($sql, $params, $types)->fetchColumn();
```

Comment:
Note that `fetchColumn()` may fail (returning `false`). Wondering if that always throws an exception, or if we can always safely cast the result to `(int)`